modules/core/src/main/scala/fs2/kafka/KafkaConsumer.scala

```diff
@@ -554,6 +549,16 @@ object KafkaConsumer {
 
       override def toString: String =
         "KafkaConsumer$" + id
+
+      // Stop polling, then stop consuming, then stop handling messages
+      private def shutdown(graceful: Boolean): F[Unit] = {
+        val shutdownOp: Fiber[F, Unit] => F[Unit] = if (graceful) _.join else _.cancel
+        shutdownOp(polls).attempt.void *> stopConsuming *> shutdownOp(actor).attempt.void
+      }
```

Comment:
This looks nice to have but probably needs more tests. In the meantime, I think we should just have `terminate` do the same thing `fiber.cancel` and `awaitTermination` do the same thing as `fiber.join`, and then think about building higher-level functionality on top of that.

Reply:
We'd also want to update the [graceful shutdown](https://github.com/fd4s/fs2-kafka/blob/series/1.x/docs/src/main/mdoc/consumers.md#graceful-shutdown) section in the docs.

Reply:
Yeah, fair point, it would need a lot more testing. However, it does seem a little odd to not run `KafkaConsumer.stopConsuming` as part of the termination methods.

For the moment I'll just get `terminate` and `awaitTerminate` to point at the old `fiber.join/cancel` methods